formula-coroutines/src/main/java/com/instacart/formula/coroutines/CoroutineAction.kt

```diff
@@ -0,0 +1,106 @@
+package com.instacart.formula.coroutines
+
+import com.instacart.formula.Action
+import com.instacart.formula.Cancelable
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ *
+ * Adapter which allows to run a coroutine as an [Action].
+ */
+class CoroutineAction<Result> @PublishedApi internal constructor(
```

Comment:
Given that coroutines are the default technology, this will likely be moved to `formula` core module in the future and might be merged with `Action` type itself.